### PR TITLE
Implement decoder codegen

### PR DESCRIPTION
**Pull request by panferovi**

### CODE REVIEW

File: isa/risc-v.yaml
Comment (panferovi) [inline isa/risc-v.yaml:1]: I'm not sure that we need all this file

File: src/decoder/decoder.rb [hunk 1]
```diff
@@ -1,56 +1,197 @@
...
+}  // namespace RISCV
+
+#endif  // GENERATED_INSTRUCTIONS_H
+
+EOT
+    instructions_file.write(instr_enum)
+    instructions_file.close
+  end
+
+  def generate_decoder(instructions)
```
Comment (panferovi) [inline src/decoder/decoder.rb:136]: This method will be implemented

File: CMakeLists.txt [hunk 2]
```diff
@@ -35,16 +35,28 @@ add_custom_target(format
     ${EMULATOR_FILES}
 )
 
+set(ISA_FILE ${CMAKE_CURRENT_SOURCE_DIR}/isa/risc-v.yaml)
+set(GEN_DIR ${CMAKE_CURRENT_BINARY_DIR}/generated)
```
Comment (OAMichael) [inline CMakeLists.txt:39]: Maybe output generated files not in build directory?

**Review by OAMichael — APPROVED** [review]
**OK**

**Review by includealex — APPROVED** [review]
@panferovi ok

**Pull request merged**